docs/requirements/ADP-957.md

```diff
@@ -0,0 +1,201 @@
+# Basic definitions
+
+*   **Collateral inputs**, in a transaction, are used to cover transaction fees
+    in the case that a non-native script fails. The term **collateral** refers
+    to the total ada contained in the UTxOs referenced by collateral inputs.
+    Collateral inputs can **only** contain ada: they may **not** contain
+    non-ada assets.
+
+*   The protocol parameter `collateralPercent` is used to specify the
+    percentage of the total transaction fee its collateral must (at a minimum)
+    cover. The collateral inputs must not themselves be locked by a script.
+
+*   The protocol parameter `maxCollateralInputs` is used to limit,
+    additionally, the total number of collateral inputs, and thus the total
+    number of additional signatures that must be checked during validation.
+
+# Requirements
+
+1.  The wallet application, when requested to create a transaction requiring
```

Comment:
It's not fully defined which transactions need collateral.
Transactions with plutus script-locked outputs need collateral.
The collateral needed is a percentage of the total of plutus script-locked outputs.